Packet++/header/GvcpLayer.h

```diff
@@ -0,0 +1,772 @@
+		{
+			return data[0] == internal::kGvcpMagicNumber;
+		};
+
+		/**
+		 * @brief Construct a new GvcpLayer object
+		 * @param[in] data A pointer to the raw data
+		 * @param[in] dataLen Size of the data in bytes
+		 * @param[in] prevLayer A pointer to the previous layer
+		 * @param[in] packet A pointer to the Packet instance where layer will be stored in
+		 * @return GvcpLayer* A pointer to the constructed GvcpLayer object
+		 */
+		static GvcpLayer* parseGvcpLayer(uint8_t* data, size_t dataLen, Layer* prevLayer, Packet* packet);
+
+	protected:
+		GvcpLayer() = default;
+
+		/**
+		 * @brief Construct a new GvcpLayer object
+		 * @param[in] data A pointer to the raw data
+		 * @param[in] dataLen Size of the data in bytes
+		 * @param[in] prevLayer A pointer to the previous layer
+		 * @param[in] packet A pointer to the Packet instance where layer will be stored in
+		 */
+		GvcpLayer(uint8_t* data, size_t dataLen, Layer* prevLayer, Packet* packet);
+
+		// implement Layer's abstract methods
+		void parseNextLayer() override
+		{}
+
+		// implement Layer's abstract methods
+		void computeCalculateFields() override
+		{}
+
+		// implement Layer's abstract methods
+		OsiModelLayer getOsiModelLayer() const override
+		{
+			return OsiModelLayer::OsiModelApplicationLayer;
+		}
+	};
+
+	/// @brief Gvcp request layer
+	class GvcpRequestLayer : public GvcpLayer
+	{
+	public:
+		/**
+		 * @brief Construct a new GvcpLayer object
+		 * @param[in] data A pointer to the raw data
+		 * @param[in] dataLen Size of the data in bytes
+		 * @param[in] prevLayer A pointer to the previous layer
+		 * @param[in] packet A pointer to the Packet instance where layer will be stored in
+		 */
+		GvcpRequestLayer(uint8_t* data, size_t dataLen, Layer* prevLayer, Packet* packet);
+
+		/**
+		 * @brief Construct a new GvcpRequestLayer object
+		 * @param[in] data A pointer to the data including the header and the payload
+		 * @param[in] dataSize The size of the data in bytes
+		 */
+		GvcpRequestLayer(const uint8_t* data, size_t dataSize);
+
+		/**
+		 * @brief Construct a new GvcpRequestLayer object
+		 * @param[in] command The command
+		 * @param[in] payloadData A pointer to the payload data, optional
+		 * @param[in] payloadDataSize The size of the payload data in bytes, optional
+		 * @param[in] flag The flag, optional
+		 * @param[in] requestId The request ID, it should be always larger than 1, optional
+		 * @note all the parameters will be converted to the network byte order
+		 */
+		explicit GvcpRequestLayer(GvcpCommand command, const uint8_t* payloadData = nullptr,
+		                          uint16_t payloadDataSize = 0, GvcpFlag flag = 0, uint16_t requestId = 1);
+
+		/**
+		 * @brief Get the header object
+		 * @return GvcpRequestHeader* A pointer to the header object
+		 */
+		GvcpRequestHeader* getGvcpHeader() const
+		{
+			return reinterpret_cast<GvcpRequestHeader*>(m_Data);  // the header is at the beginning of the data
+		}
```

Comment:
this method can probably be private/protected

Reply:
Please see this comment: https://github.com/seladb/PcapPlusPlus/pull/1357#discussion_r1821925821

Reply:
fixed.